faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -494,23 +494,23 @@ private void handleQueryErrors(int statusCode, String body) {
           }
         }
 
-        HttpResponses.QueryErrorResponse errorResponse = HttpResponses.QueryErrorResponse.create(statusCode, parsedErrors);
-
         switch (statusCode) {
           case 400:
-            throw new BadRequestException(errorResponse);
-          case 401:
-            throw new UnauthorizedException(errorResponse);
           case 403:
-            throw new PermissionDeniedException(errorResponse);
           case 404:
-            throw new NotFoundException(errorResponse);
+            throwQueryError(parsedErrors, statusCode);
+          case 401:
+            throw new UnauthorizedException("Unauthorized", statusCode);
+          case 440:
+            throw new ProcessingTimeLimitExceededException("Processing timeout exceeded", statusCode);
           case 500:
-            throw new InternalException(errorResponse);
+            throw new InternalException("Internal Server Error", statusCode);
```

Comment:
We still need to return as much info as possible from the original response as possible here, as it helps with debugging if we ever get a report. If parsing the response body fails, we can fall back to a generic version.